core/src/main/clojure/xtdb/core/datalog.clj

```diff
@@ -31,7 +31,8 @@
 (s/def ::form
   (s/or :logic-var ::logic-var
         :fn-call ::fn-call
-        :literal ::value))
+        :literal ::value
+        :sub-query ::sub-query))
```

Comment:
worth putting `:sub-query` ahead of `:fn-call`? think `:fn-call` might swallow a sub-q